### PR TITLE
initial conversion to OVOS plugin

### PR DESCRIPTION
**Pull request by JarbasAl**
this is a work in progress PR, i removed everything and stripped the plugin to the essential as if it was a OVOS native plugin

now i will re-add the stuff that should not have been removed for neon_utils etc

ignore the rebranding on the setup.py

### CODE REVIEW

**Comment by JarbasAl** [comment]
added stopwatch to ovos_utils, seems like it was missing https://github.com/OpenVoiceOS/ovos_utils/pull/78

added metrics capability directly into ovos plugin manager, this introduces a new handle_metric method that can be subclassed in TTS, i am just making it log the metrics for now https://github.com/OpenVoiceOS/OVOS-plugin-manager/commit/b5eb94637b637b32388fbe6152d26f851b72df62

**Comment by JarbasAl** [comment]
also requires https://github.com/NeonJarbas/neon_audio/commit/f3e7363bcea684159c9bfaae1af2ffeb809a497b

**Comment by JarbasAl** [comment]
@NeonDaniel this has been modernized to latest best practices

Note the changes in the package and plugin name as this might be breaking compat

This should now be ready to be made the default replacement for the mycroft tts engine in ovos. Will make a follow up PR in OPM once this is merged

Context: in ovos-core we removed all hardcoded TTS engines and need to have mappings for an equivalent default plugin available for backwards compat

File: neon_tts_plugin_mozilla_remote/__init__.py [hunk 1]
```diff
@@ -19,54 +19,32 @@
...
-                with open(wav_file, "wb") as f:
-                    f.write(wav_data)
-            LOG.debug(f"File access time={stopwatch.time}")
+        super().__init__(lang, config,
+                         RemoteMozillaTTSValidator(self),
+                         audio_ext="mp3")
+        self.url = self.config.get("api_url") or \
+                   self.config.get("url", "http://0.0.0.0:5002/api/tts")  # mycroft-core compat
+
+    def get_tts(self, sentence, wav_file):
```
Comment (NeonDaniel) [inline neon_tts_plugin_mozilla_remote/__init__.py]: Should this accept args/kwargs for backwards compat?

File: setup.py [hunk 2]
```diff
@@ -55,15 +55,15 @@ def get_requirements(requirements_filename: str):
                 version = line.split("'")[1]
 
 setup(
-    name='neon-tts-plugin-mozilla_remote',
+    name='neon-tts-plugin-mozilla-remote',
```
Comment (NeonDaniel) [inline setup.py]: Is this change necessary? It seems that it would create a new package on PyPI

Comment (JarbasAl) [reply setup.py]: this was just to keep the naming convention in plugins + the convention of matching plugin name to package name

but can be reverted to not create a new package, those are just guidelines not required!

File: neon_tts_plugin_mozilla_remote/__init__.py (hunk 1 above)
Comment (JarbasAl) [reply neon_tts_plugin_mozilla_remote/__init__.py]: no need, only those 2 args are part of the official api, other args are optional

execute and synth parse kwarsg and signature internally and only pass the ones accepted by this method, direct usage can be a concern however.... but thats breaking the api? bump a version is enough?

Comment (NeonDaniel) [reply neon_tts_plugin_mozilla_remote/__init__.py]: Ah, right. I forgot we are patching this differently. Yes, a major version bump will address it

File: setup.py (hunk 2 above)
Comment (NeonDaniel) [reply setup.py]: Okay, let's revert this one and try to keep up the standards in the future

File: setup.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 from setuptools import setup, find_packages
 from os import path, getenv
 
-PLUGIN_ENTRY_POINT = 'mozilla_remote = neon_tts_plugin_mozilla_remote:MozillaRemoteTTS'
```
Comment (NeonDaniel) [inline setup.py]: This is a breaking change
